src/figuresClasses.ts

```diff
@@ -1,19 +1,107 @@
-export interface Figure {
+type Shape ='triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape;
+  color: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle';
 
-}
+  color: string;
+
+  a: number;
+
+  b: number;
+
+  c: number;
 
-export class Circle {
+  private halfSides: number;
 
+  constructor(color: string, a: number, b: number, c: number) {
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+    this.halfSides = (this.a + this.b + this.c) / 2;
+    this.checkSides();
+  }
+
+  private checkSides(): void {
+    if (this.a + this.b <= this.c
+      || this.b + this.c <= this.a
+      || this.c + this.a <= this.b) {
+      throw new Error('That is not a triagnle!');
+    }
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
```

Comment:
just run this code inside the constructor, no need to create a function as you won't reuse it anyway.
Also, you don't have to use this.a, you can use just `a` as you want to only check the coming variables